locations/spiders/tjmaxx.py

```diff
@@ -0,0 +1,131 @@
+        )
+
+
+    def process_hours(self, hours):
+        """ This should capture these time formats as on tjmaxx
+                # Mon-Wed: 9:30a-9:30p,
+                # Thanksgiving Day: CLOSED,  <- this gets appended without modification
+                # Fri-Sat: 7a-10p,
+                # Sun: 9a-10p
+        :param hours:
+        :return:  string - in this format Mo-Th 11:00-12:00; Fr-Sa 11:00-01:00;
+        """
+        working_hour = []
+
+        for t in hours.split(","):
+            match = re.search(r"^((\w{2})\w(?:-(\w{2})\w)?\:\s(\d+)(\:\d+)?(\w)-(\d+)(\:\d+)?(\w))$",
+                              t.strip(),
+                              re.IGNORECASE)
+            if match:
+                m = list(match.groups())
+                if len(m) == 9:
+                    # replace second entry for day of week for formats like Sun: 9a-12p
+                    if m[2] is None:
+                        m[2] = ""
+                    else:
+                        m[2] = "-" + m[2]
+
+                    # concatenate to our format
+                    if m[4] is None:
+                        if m[7] is None:
+                            # time is like Mon-Wed: 9a-9p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + ":00-" +
+                                                str(int(m[6]) + self.am_pm(m[6], m[8])) + ":00")
+                        else:
+                            # time is like Mon-Wed: 9a-9:30p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + ":00-" +
+                                                str(int(m[6]) + self.am_pm(m[6], m[8])) + m[7])
+                    else:
+                        if m[7] is None:
+                            # time is like Mon-Wed: 9:30a-9p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + m[4] +
+                                                "-" + str(int(m[6]) + self.am_pm(m[6], m[8])) + ":00")
+                        else:
+                            # time is like Mon-Wed: 9:30a-9:30p
+                            working_hour.append(m[1] + m[2] + " " + str(int(m[3]) + self.am_pm(m[3], m[5])) + m[4] +
+                                                "-" + str(int(m[6]) + self.am_pm(m[6], m[8])) + m[7])
+            # else:
+            #     working_hour.append(t)
+
+        if len(working_hour):
+            return "; ".join(working_hour)
+        else:
+            # if something fails, return original date
+            return hours
+
+    def am_pm(self, m3, x):
+        """
+            A convenience method to fix noon and midnight issues
+        :param m3: the hour has to be passed it to accurately decide 12noon and midnight
+        :param x: this is either a or p i.e am pm
+        :return: the hours that must be added
+
+        """
+        diff = 0
+        if x == 'a':
```

Comment:
I don't think the `am_pm()` function is working as expected. For example, [this store](https://tjmaxx.tjx.com/store/stores/Peoria-IL-61614/309/aboutstore) has these hours:

```
Mon-Fri: 9a-10p
Sat-Sun: 8a-10p
```

but the output is getting `Mo-Fr 9:00-10:00; Sa-Su 8:00-10:00` instead of `Mo-Fr 09:00-22:00; Sa-Su 08:00-22:00`.

Reply:
Thanks for these notes.